pkgs/by-name/el/eliza/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  clang,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "eliza";
+  version = "0-unstable-2025-02-21";
+  src = fetchFromGitHub {
+    owner = "anthay";
+    repo = "ELIZA";
+    rev = "27bcf6e5fb1d32812cc0aab8133fa5e395d41773";
+    hash = "sha256-/i8mckRQWTK1yI/MNaieSuE+dx94DMdrABkqf/bXQbM=";
+  };
+
+  nativeBuildInputs = [
+    clang
+  ];
+
+  doCheck = true;
+
+  # Unit tests are executed automatically on execution
+  checkPhase = ''
+    runHook preCheck
+    ./eliza --help
+    runHook postCheck
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    clang++ -std=c++20 -pedantic -o eliza ./src/eliza.cpp
```

Comment:
Maybe don't hardcode clang. Will the package build with GCC? If so, please just use $CC so that we don't have to pull Clang/LLVM when we don't have to.